datasets/timit_asr/timit_asr.py

```diff
@@ -0,0 +1,171 @@
+# coding=utf-8
+# Copyright 2021 The TensorFlow Datasets Authors and the HuggingFace Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""TIMIT automatic speech recognition dataset."""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+
+import pandas as pd
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{
+  title={TIMIT Acoustic-Phonetic Continuous Speech Corpus},
+  author={Garofolo, John S., et al},
+  ldc_catalog_no={LDC93S1},
+  DOI={https://doi.org/10.35111/17gk-bn40},
+  journal={Linguistic Data Consortium, Philadelphia},
+  year={1983}
+}
+"""
+
+_DESCRIPTION = """\
+The TIMIT corpus of reading speech has been developed to provide speech data for acoustic-phonetic research studies
+and for the evaluation of automatic speech recognition systems.
+
+TIMIT contains high quality recordings of 630 individuals/speakers with 8 different American English dialects,
+with each individual reading upto 10 phonetically rich sentences.
+
+More info on TIMIT dataset can be understood from the "README" which can be found here:
+https://catalog.ldc.upenn.edu/docs/LDC93S1/readme.txt
+"""
+
+_URL = "https://data.deepai.org/timit.zip"
+
+
+class TimitASRConfig(datasets.BuilderConfig):
+    """BuilderConfig for TimitASR."""
+
+    def __init__(self, **kwargs):
+        """
+        Args:
+          data_dir: `string`, the path to the folder containing the files in the
+            downloaded .tar
+          citation: `string`, citation for the data set
+          url: `string`, url for information about the data set
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(TimitASRConfig, self).__init__(version=datasets.Version("2.0.1", ""), **kwargs)
+
+
+class TimitASRConfig(datasets.GeneratorBasedBuilder):
```

Comment:
Small class name issue here:
```suggestion
class TimitASR(datasets.GeneratorBasedBuilder):
```
note that after this change you'll need to regenerate the dataset_infos.json file